src/aal/aal_cheri_mips.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#define SNMALLOC_VA_BITS_64
+
+extern "C"
+{
+#include <machine/cheri.h>
+}
+
+namespace snmalloc
+{
+  /**
+   * CHERI-MIPS specific architecture abstraction layer.
+   */
+  class AAL_CHERI_MIPS
+  {
+    static inline void halt_out_of_order()
+    {
+      static unsigned st;
+      __sync_swap(&st, 0); // XXX: Overkill!
```

Comment:
This (and the use below) look as if they're relying on a not-very-well-specified builtin emitting a particular barrier on MIPS.  I think I'd rather have inline assembly than this.  Alternatively, is one of the `std::thread_fence` variants sufficient?

Reply:
I'll check with Alex and Jon.